operator/pkg/ciliumendpointslice/cescache.go

```diff
@@ -74,6 +101,65 @@ func (c *CESCache) deleteNode(nodeName NodeName) []CESKey {
 	return nil
 }
 
+// Insert a CID into the cache and return the CESs which need to be reconciled
+func (c *CESCache) insertCID(cid CID, gidLabels Labels) []CESKey {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	var cesToReconcile []CESKey
+	// Clean up old state
+	if oldLabels, ok := c.cidToGidLabels[cid]; ok && oldLabels != gidLabels {
+		cidSet := c.globalIdLabelsToCIDSet[oldLabels]
+		cidSet.ids.Delete(cid)
+		// If the selectedID is the same as the CID being removed, update it to another valid CID, if it exists.
+		if cidSet.selectedID == cid {
+			changed := cidSet.setSelectedIDLocked("")
+			if changed {
+				cesToReconcile = append(cesToReconcile, c.getCESForCEPs(cidSet.ceps)...)
```

Comment:
`s/getCESForCEPs/getCESForCEPsLocked` (and similarly for a couple of other occurrences in this commit)